src/hssm/register.py

```diff
@@ -0,0 +1,96 @@
+"""Module for registering custom models in HSSM."""
+
+from pprint import pformat, pp
+from typing import cast
+
+from .defaults import (
+    DefaultConfig,
+    LoglikConfigs,
+    SupportedModels,
+)
+from .defaults import (
+    default_model_config as registered_models,
```

Comment:
This is not exactly lazy. Once `default_model_config` is imported, the entire dictionary is loaded into memory.

One thing that we can make the loading process lazy is to separate the dictionary into variables with the same name as the model, and then we can dynamically import them as needed. Check out [`importlib.import_module`](https://docs.python.org/3/library/importlib.html#importlib.import_module)

There are definitely other ways in which we can do this. Feel free to use whatever you think works best